stan/math/fwd/fun/polar.hpp

```diff
@@ -0,0 +1,56 @@
+#ifndef STAN_MATH_FWD_FUN_POLAR_HPP
+#define STAN_MATH_FWD_FUN_POLAR_HPP
+
+#include <stan/math/fwd/core.hpp>
+#include <stan/math/fwd/meta.hpp>
+#include <stan/math/prim/fun/polar.hpp>
+#include <complex>
+#include <type_traits>
+
+namespace stan {
+namespace math {
+
+/**
```

Comment:
Yes, it does create ambiguities.  The problem here is the built-in 

```
template<class T> complex<T> polar (const T& rho, const T& theta = 0);
```

It's not general enough for all of our autodiff types (or even for `int` and `double`) but is just general enough to cause ambiguities with perfect matches when the types agree.

Some of these cases can be handled by specializations that work in gcc and clang, but only a subset because gcc and clang don't have the same built-in templates.

I chose to handle every function uniformly with simple autodiff overloads that would unambiguosly match better than any fully templated version.   That's not the way I did it the first or second or third times, but in the end, it was just so much simpler to have everything done the same way. This also future-proofs us to some extent against future fixes to `std::complex` in gcc and clang.